core/src/core/usb.zig

```diff
@@ -530,7 +530,7 @@ pub const EndpointDescriptor = extern struct {
 
     pub fn serialize(self: *const @This()) [7]u8 {
         var out: [7]u8 = undefined;
-        out[0] = 7; // length
+        out[0] = self.length;
```

Comment:
This function should assert `self.length` to be 7, and hardcode it with either `@sizeOf(@This())` or use `out.len`.

Otherwise, a user could pass in a bad `self` which has length = 6 or similar.

Reply:
That a pretty good idea, will implement it.